assets/tests/components/mapPage/mapDisplay.test.tsx

```diff
@@ -322,7 +388,111 @@ describe("<MapPage />", () => {
           expect(
             screen.queryAllByRole("button", { name: /^run/ })
           ).toHaveLength(0)
-          expect(getVehiclePropertiesCard()).toBeVisible()
+          expect(vehiclePropertiesCard.get()).toBeVisible()
+        })
+      })
+
+      describe("selection is a route pattern", () => {
+        test("RPC doesn't display if route data hasn't loaded yet", () => {
+          setHtmlWidthHeightForLeafletMap()
+
+          const route = routeFactory.build()
+          const vehicles = randomLocationVehicle.buildList(3)
+
+          mockUseVehiclesForRouteMap({ [route.id]: vehicles })
+          const routePattern = routePatternFactory.build({ routeId: route.id })
+          ;(usePatternsByIdForRoute as jest.Mock).mockReturnValue({
+            [routePattern.id]: routePattern,
+          })
+
+          render(
+            <RoutesProvider routes={[]}>
+              <MapDisplay
+                selectedEntity={{
+                  type: SelectedEntityType.RoutePattern,
+                  routeId: route.id,
+                  routePatternId: routePattern.id,
+                }}
+                setSelection={jest.fn()}
+                showSelectionCard={true}
+              />
+            </RoutesProvider>
+          )
+
+          expect(routePropertiesCard.query()).not.toBeInTheDocument()
+        })
+
+        test("RPC doesn't display if selected pattern ID doesn't match the route patterns", () => {
+          setHtmlWidthHeightForLeafletMap()
+
+          const route = routeFactory.build()
+          const vehicles = randomLocationVehicle.buildList(3)
+
+          mockUseVehiclesForRouteMap({ [route.id]: vehicles })
+          const routePattern = routePatternFactory.build({ routeId: route.id })
+          ;(usePatternsByIdForRoute as jest.Mock).mockReturnValue({
+            [routePattern.id]: routePattern,
+          })
+          render(
+            <RoutesProvider routes={[route]}>
+              <MapDisplay
+                selectedEntity={{
+                  type: SelectedEntityType.RoutePattern,
+                  routeId: route.id,
+                  routePatternId: "otherRoutePatternId",
+                }}
+                setSelection={jest.fn()}
+                showSelectionCard={true}
+              />
+            </RoutesProvider>
+          )
+
+          expect(routePropertiesCard.query()).not.toBeInTheDocument()
+        })
+        test("clicking vehicle route dispatches setSelection event", async () => {
```

Comment:
**suggestion:** the "clicking vehicle route..." part of the title is a little confusing compared to what the test actually does, maybe "selecting pattern in RPC dispatches setselection event" or something like that would be better?